### PR TITLE
Fixed #25335 -- Fixed regression where QuerySet.iterator() didn't return an iterator.

### PR DESCRIPTION
**Pull request by gavinwahl**
https://code.djangoproject.com/ticket/25335

### CODE REVIEW

File: tests/lookup/tests.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from datetime import datetime
 from operator import attrgetter
 from unittest import skipUnless
+import collections
```
Comment (timgraham) [inline tests/lookup/tests.py]: This is sorted above the "from" imports.

**Comment by timgraham** [comment]
It seems more tests are needed to cover all the changes, no?

**Comment by gavinwahl** [comment]
I don't know, if there are tests missing now they were missing in 4c3bfe90 too.

**Comment by timgraham** [comment]
Oh, I was looking at the `ValuesIterable` and similar changes, but those are just renames it seems.

**Comment by loic** [comment]
In term of test, I guess an assertion that it now returns an iterator wouldn't hurt.

File: tests/lookup/tests.py [hunk 2]
```diff
@@ -75,6 +76,8 @@ def test_lookup_date_as_str(self):
     def test_iterator(self):
         # Each QuerySet gets iterator(), which is a generator that "lazily"
         # returns results using database-level iteration.
+        self.assertIsInstance(Article.objects.iterator(), collections.Iterator)
```
Comment (timgraham) [inline tests/lookup/tests.py]: Like this one?

Comment (loic) [inline tests/lookup/tests.py]: Hehe yeah, I overlooked it.

File: tests/lookup/tests.py (hunk 1 above)
Comment (gavinwahl) [inline tests/lookup/tests.py]: Fixed

**Comment by timgraham** [comment]
merged in 627c7eb7bf81fe4faac552475fa13f449dc8dd33, thanks!

**Pull request closed without merging**